ratelimit_test.go

```diff
@@ -29,36 +28,60 @@ type runner interface {
 type runnerImpl struct {
 	t *testing.T
 
-	clock *clock.Mock
-	count atomic.Int32
+	clock       *clock.Mock
+	constructor func(int, ...Option) Limiter
+	count       atomic.Int32
 	// maxDuration is the time we need to move into the future for a test.
 	// It's populated automatically based on assertCountAt/afterFunc.
 	maxDuration time.Duration
 	doneCh      chan struct{}
 	wg          sync.WaitGroup
 }
 
-func runTest(t *testing.T, fn func(runner)) {
-	r := runnerImpl{
-		t:      t,
-		clock:  clock.NewMock(),
-		doneCh: make(chan struct{}),
+func runTest(t *testing.T, fn func(testRunner)) {
+	implementations := []struct {
```

Comment:
minor: This is a bit long. Consider renaming this variable to "tests" to match the usual name for table tests, or "impls" if you think that wouldn't be appropriate since these are not necessarily tests.